### PR TITLE
Graupner Targets

### PR DESCRIPTION
**Pull request by Scavanger**
On special request from @Ralfde and Graupner:

Allow up to 12 servos for bigger airplanes: 
new targets: 
GRAUPNERF413: 1 motor, 12 servos, UARTs: 1, 3 shared with I2C, 6, softserial 
GRAUPNERF4V6: 2 motors, 8 servos 
GRAUPNERF4V3: 1 motor at PPM pin, 6 servos 
GRAUPNERF7NXT: 2 motors, 8 servos

### CODE REVIEW

**Comment by DzikuVx** [comment]
@Scavanger please read https://github.com/iNavFlight/inav/blob/master/docs/policies/NEW_HARDWARE_POLICY.md

File: src/main/drivers/pwm_mapping.h [hunk 1]
```diff
@@ -23,15 +23,15 @@
 
 #if defined(TARGET_MOTOR_COUNT)
 #define MAX_PWM_MOTORS TARGET_MOTOR_COUNT
-#define MAX_PWM_SERVOS 8
+#define MAX_PWM_SERVOS 12
```
Comment (DzikuVx) [inline src/main/drivers/pwm_mapping.h:26]: There already is #4868 that changes that

File: src/main/target/FIREWORKSV2/target.c [hunk 2]
```diff
@@ -38,22 +38,65 @@
...
+	DEF_TIM(TIM4, CH1, PB6, TIM_USE_MC_SERVO | TIM_USE_FW_SERVO,    0, 0), // S10_OUT LED Strip
+	#elif GRAUPNERF413
+	DEF_TIM(TIM2, CH3, PA2, TIM_USE_MC_MOTOR | TIM_USE_FW_SERVO,    0, 0),  // S7_OUT TX2	TIM5_CH3	TIM9_CH1
+	DEF_TIM(TIM2, CH1, PA0, TIM_USE_MC_MOTOR | TIM_USE_FW_SERVO,    0, 0),  // S8_OUT TX4_AIRSPEED_RSSI	TIM5_CH1
+	DEF_TIM(TIM4, CH2, PB7, TIM_USE_MC_SERVO | TIM_USE_FW_SERVO,    0, 0), // S10_OUT Camera Control
+	DEF_TIM(TIM4, CH1, PB6, TIM_USE_MC_SERVO | TIM_USE_FW_SERVO,    0, 0), // S11_OUT LED Strip
+	
+	DEF_TIM(TIM4, CH3, PB8, TIM_USE_MC_SERVO | TIM_USE_FW_SERVO,    0, 0),	// S12 SCL
+	DEF_TIM(TIM4, CH4, PB9, TIM_USE_MC_SERVO | TIM_USE_FW_SERVO,    0, 0),	// S13 SDA
+//	DEF_TIM(TIM3, CH1, PC6,                    TIM_USE_FW_SERVO,    0, 0),	// S14 TX6
```
Comment (DzikuVx) [inline src/main/target/FIREWORKSV2/target.c]: used or not?

File: src/main/target/FIREWORKSV2/target.c [hunk 3]
```diff
@@ -38,22 +38,65 @@
...
+
+	DEF_TIM(TIM5, CH2, PA1,                    TIM_USE_FW_MOTOR,    0, 0),  // S9_OUT RX4
+	
+//  changed files:
+//	src/main/drivers/pwm_mapping.h line 26-34
```
Comment (DzikuVx) [inline src/main/target/FIREWORKSV2/target.c:89]: Not needed comment

File: src/main/target/OMNIBUSF4/target.c [hunk 4]
```diff
@@ -26,31 +26,52 @@
 const timerHardware_t timerHardware[] = {
 #if defined(OMNIBUSF4PRO) || defined(OMNIBUSF4V3)
     DEF_TIM(TIM10, CH1, PB8,  TIM_USE_PPM,                  0, 0), // PPM
-    DEF_TIM(TIM4,  CH4, PB9,  TIM_USE_ANY,                  0, 0), // S2_IN
+    DEF_TIM(TIM4,  CH4, PB9,  TIM_USE_ANY,                  0, 0), // S2_IN		
+#elif defined(GRAUPNERF4V3)
+//    DEF_TIM(TIM10, CH1, PB8,  TIM_USE_MC_SERVO | TIM_USE_FW_MOTOR, 0, 0), // PPM
```
Comment (DzikuVx) [inline src/main/target/OMNIBUSF4/target.c]: Why those are commented out?

File: src/main/target/OMNIBUSF4/target.c [hunk 5]
```diff
@@ -26,31 +26,52 @@
...
+#ifndef GRAUPNERF4V3
+   DEF_TIM(TIM1,  CH1, PA8,  TIM_USE_MC_MOTOR | TIM_USE_MC_SERVO | TIM_USE_FW_SERVO,   0, 0), // S6_OUT
+#endif
 #if (defined(OMNIBUSF4PRO) || defined(OMNIBUSF4V3)) && !defined(OMNIBUSF4PRO_LEDSTRIPM5)
     DEF_TIM(TIM4,  CH1, PB6,  TIM_USE_LED,                                              0, 0), // LED strip for F4 V2 / F4-Pro-0X and later (RCD_CS for F4)
 #endif
+	#ifdef GRAUPNERF4V3
+	DEF_TIM(TIM1,  CH1, PA8,  TIM_USE_MC_MOTOR | TIM_USE_MC_SERVO | TIM_USE_FW_MOTOR,   0, 0), // S6_OUT
+	DEF_TIM(TIM4,  CH3, PB8,  TIM_USE_MC_SERVO | TIM_USE_FW_SERVO, 0, 0), // PPM	TIM4_CH3	TIM10_CH1
+//    DEF_TIM(TIM4,  CH3, PB8,  TIM_USE_MC_SERVO | TIM_USE_FW_MOTOR, 0, 0), // PPM	TIM4_CH3	TIM10_CH1
```
Comment (DzikuVx) [inline src/main/target/OMNIBUSF4/target.c]: Please remove

File: src/main/target/OMNIBUSF7NXT/target.c [hunk 6]
```diff
@@ -40,12 +40,37 @@ const timerHardware_t timerHardware[] = {
     // OUTPUT 1-4
     DEF_TIM(TIM3, CH2, PB5, TIM_USE_MC_MOTOR | TIM_USE_FW_SERVO,    1, 0),  // D(1, 5, 5)
     DEF_TIM(TIM3, CH1, PB4, TIM_USE_MC_MOTOR | TIM_USE_FW_SERVO,    1, 0),  // D(1, 4, 5)
-    DEF_TIM(TIM3, CH3, PB0, TIM_USE_MC_MOTOR | TIM_USE_FW_SERVO,    1, 0),  // D(1, 7, 5)
+//    DEF_TIM(TIM3, CH3, PB0, TIM_USE_MC_MOTOR | TIM_USE_FW_SERVO,    1, 0),  // D(1, 7, 5)
```
Comment (DzikuVx) [inline src/main/target/OMNIBUSF7NXT/target.c]: Please remove unused and commented timer definitions is this file too

**Review by DzikuVx — CHANGES_REQUESTED** [review]

**Comment by digitalentity** [comment]
Those targets need some serious cleanup before they could be considered. Now they have a lot of unneeded artifacts and are quite a pain to maintain.

**Comment by Ralfde** [comment]
All Graupner targets use Omnibus hardware, just software is changed for Graupner airplane customers:
GRAUPNERF413: Omnibus F4V6 (second accel/gyro will be added
GRAUPNERF4V6 = Omnibus F4V6
GRAUPNERF4V3 = Omnibus F4V3
GRAUPNERF7NXT = Omnibus F7NXT
Our airplane customers need more servos for the airplanes and this is why I made new targets.
In case we should offer just custom builds to our customers, we could do so, except of the GRAUPNERF413, which should get a new layout, soon, so the user can better use it for airplanes.

Not needed command lines could be deleted.

**Comment by Ralfde** [comment]
The cleaned code:
[scr.zip](https://github.com/iNavFlight/inav/files/3333092/scr.zip)

**Comment by Ralfde** [comment]
[servos.zip](https://github.com/iNavFlight/inav/files/3398473/servos.zip)
I think this should solve the conflict.

**Comment by Ralfde** [comment]
[servos.zip](https://github.com/iNavFlight/inav/files/3398473/servos.zip)
I think this should solve the conflict.

**Comment by digitalentity** [comment]
Sorry, this fell through the cracks. @Ralfde is this still relevant?

**Comment by digitalentity** [comment]
@Scavanger can you amend the PR?

**Comment by Ralfde** [comment]
Found some wrong definitions in the GRAUPNERF413 target for the I2C.
Please use the new file for the FIREWORKSV2 TARGET:
[target.zip](https://github.com/iNavFlight/inav/files/3613554/target.zip)

As all other changes seem to be already in the development tree, except the new targets, I think it's maybe better to split the Graupner targets in seperate target folders to have it seperated and a better overview in the code. If you would prefer this, too, please use this new targets instead of the changes in the other targets:
I compiled them already with the latest development code and no error occured.
[target.zip](https://github.com/iNavFlight/inav/files/3613563/target.zip)

**Pull request closed without merging**